doc/source/installation/docker.rst

```diff
@@ -0,0 +1,159 @@
+Using a deal.II  Docker container
+=================================
+
+The deal.II library is a mature project containing `O(1,000,000)` lines of C++ code, itself depending on other similar libraries (e.g. Trilinos). Compiling deal.II with its exact dependencies can be a daunting task even for moderate C++ programmers - one with many knobs and dials, normally taking 4-6 hours.
+
+Thankfully, the deal.II dev heroes also maintain some `Docker Hub <https://hub.docker.com/r/dealii/dealii/>`_ images containing pre-built deal.II libraries in a Ubuntu environment. For those unfamiliar with Docker, it allows the creation of completely isolated environments, most often pre-configured for specific apps; it is almost like a virtual machine, but running natively on your current OS (the technical term is *OS-level* virtualization, rather than *hardware-level*) - so scientific programs shouldn't have any degradation in performance. 
+
+Below are some instructions for getting pre-configured deal.II environments, reducing the time needed to install lethe's dependencies from hours to minutes. There are 3 main parts in this tutorial:
+
+- If you have experience with Docker, the **0. TL;DR** section should be enough.
+- The next two sections have a brief introduction to Docker with enough information to get you started with deal.II containers.
+- The final **3. Saving Docker Commands in a Bash Script** section has a shell script you can run directly to start and manage deal.II containers.
+
+Sometimes it's easier to effectively set up a completely new OS than compile millions of lines of C++...
+
+
+0. TL;DR
+--------
```

Comment:
```suggestion

.. _tl-dr:

0. TL;DR
--------
```

Reply:
You can actually refer to subsections directly with hyperlinks in Sphinx, this is slightly cleaner than just using bold ;).